cmd/aws-vpc-cni/main.go

```diff
@@ -0,0 +1,410 @@
+	return json.Unmarshal([]byte(inFile), &result)
+}
+
+func generateJSON(jsonFile string, outFile string) error {
+	byteValue, err := ioutil.ReadFile(jsonFile)
+	if err != nil {
+		return err
+	}
+
+	var nodeIP string
+	nodeIP, err = getNodePrimaryV4Address()
+	if err != nil {
+		log.Errorf("Failed to get Node IP")
+		return err
+	}
+
+	vethPrefix := getEnv(envVethPrefix, defaultVethPrefix)
+	mtu := getEnv(envEniMTU, defaultMTU)
+	podSGEnforcingMode := getEnv(envPodSGEnforcingMode, defaultPodSGEnforcingMode)
+	pluginLogFile := getEnv(envPluginLogFile, defaultPluginLogFile)
+	pluginLogLevel := getEnv(envPluginLogLevel, defaultPluginLogLevel)
+	egressV4pluginLogFile := getEnv(envEgressV4PluginLogFile, defaultEgressV4PluginLogFile)
+	enabledIPv6 := getEnv(envEnIPv6, defaultEnableIPv6)
+	randomizeSNAT := getEnv(envRandomizeSNAT, defaultRandomizeSNAT)
+
+	netconf := string(byteValue)
+	netconf = strings.Replace(netconf, "__VETHPREFIX__", vethPrefix, -1)
+	netconf = strings.Replace(netconf, "__MTU__", mtu, -1)
+	netconf = strings.Replace(netconf, "__PODSGENFORCINGMODE__", podSGEnforcingMode, -1)
+	netconf = strings.Replace(netconf, "__PLUGINLOGFILE__", pluginLogFile, -1)
+	netconf = strings.Replace(netconf, "__PLUGINLOGLEVEL__", pluginLogLevel, -1)
+	netconf = strings.Replace(netconf, "__EGRESSV4PLUGINLOGFILE__", egressV4pluginLogFile, -1)
+	netconf = strings.Replace(netconf, "__EGRESSV4PLUGINENABLED__", enabledIPv6, -1)
+	netconf = strings.Replace(netconf, "__RANDOMIZESNAT__", randomizeSNAT, -1)
+	netconf = strings.Replace(netconf, "__NODEIP__", nodeIP, -1)
+
+	byteValue = []byte(netconf)
+
+	enBandwidthPlugin := getEnv(envEnBandwidthPlugin, "false")
+	if enBandwidthPlugin == "true" {
+		data := NetConfList{}
+		err = json.Unmarshal(byteValue, &data)
+		if err != nil {
+			return err
+		}
+
+		bwPlugin := NetConf{
+			Type:         "bandwidth",
+			Capabilities: map[string]bool{"bandwidth": true},
+		}
+		data.Plugins = append(data.Plugins, &bwPlugin)
+		byteValue, err = json.MarshalIndent(data, "", "  ")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = isValidJSON(string(byteValue))
+	if err != nil {
+		log.Fatalf("%s is not a valid json object\nerror: %s", netconf, err)
+	}
+
+	err = ioutil.WriteFile(outFile, byteValue, 0644)
+	return err
+}
+
+func validateEnvVars() bool {
+	pluginLogFile := getEnv(envPluginLogFile, defaultPluginLogFile)
+	if pluginLogFile == "stdout" {
+		log.Errorf("AWS_VPC_K8S_PLUGIN_LOG_FILE cannot be set to stdout")
+		return false
+	}
+
+	vethPrefix := getEnv(envVethPrefix, defaultVethPrefix)
+	if len(vethPrefix) > 4 {
+		log.Errorf("AWS_VPC_K8S_CNI_VETHPREFIX cannot be longer than 4 characters")
+		return false
+	}
+
+	if vethPrefix == "eth" || vethPrefix == "lo" || vethPrefix == "vlan" {
+		log.Errorf("AWS_VPC_K8S_CNI_VETHPREFIX cannot be set to reserved values 'eth', 'vlan', or 'lo'")
+		return false
+	}
+
+	podSGEnforcingMode := getEnv(envPodSGEnforcingMode, defaultPodSGEnforcingMode)
```

Comment:
Should this check be paired with `ENABLE_POD_ENI`? Why make this a mandatory check for all modes?

Reply:
I was just copying the check from the existing `entrypoint` script, but yeah, makes sense to check this only when `ENABLE_POD_ENI` is set.